launch/launch/substitutions/boolean_substitution.py

```diff
@@ -172,21 +172,26 @@ class AnySubstitution(Substitution):
     If none of the arguments evaluate to true, then this substitution returns the string 'false'.
     """
 
-    def __init__(self, *args: SomeSubstitutionsType) -> None:
+    def __init__(
+        self, *args: SomeSubstitutionsType, container: List[SomeSubstitutionsType] | None = None,
```

Comment:
It should probably be an `Iterable` because technically `parse()` gets an `Iterable[SomeSubstitutionsType]`. And I think this change makes the line go over 100 chars, so:


```suggestion
        self,
        *args: SomeSubstitutionsType,
        container: Iterable[SomeSubstitutionsType] | None = None,
```